CONTRIBUTING.md

```diff
@@ -62,3 +110,23 @@ Thanks for taking the plunge!
   rather than `[1., 2.]`
 * In docstrings, optional arguments, including separators and spaces, are surrounded by brackets,
   e.g. `mymethod(required[, optional1[, optional2] ]; kwargs...)`
+
+## Git Recommendations For Pull Requests
+
+* Avoid working from the `main` branch of your fork, creating a new branch will make it easier if DataFrame's `main` branch changes and you need to update your pull request.
+
+* Try to [squash](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) together small commits that make repeated changes to the same section of code so your pull request is easier to review, and DataFrame's won't have any broken intermediate commits. A reasonable number of separate well-factored commits is fine, especially for larger changes.
```

Comment:
there is no need to do it - we will squash the PR when merging it anyway. The request is rather: run tests of your code before sending any commit to GitHub. Only push changes when the tests of the change are passing locally.

The point is that if you send e.g. 10 commits to GitHub in one push operation it is not a problem as CI will be run only once then.